src/Histogram.cpp

```diff
@@ -59,12 +71,33 @@ int64_t shasta::Histogram2::findIndex(double x){
 void shasta::Histogram2::update(double x) {
     auto index = findIndex(x);
 
-    if (index >= 0) {
+    if (dynamicBounds){
+        // If the found index is too large, extend the end of the deque
+        if (index > histogram.size()){
+            size_t newBins = size_t(index)-histogram.size();
```

Comment:
We have been trying to avoid `size_t` in new code because it has a different size on macOS and its use tends to cause compilation problems on macOS. Please convert to `uint64_t` here and elsewhere.